test/test_testing.py

```diff
@@ -5561,11 +5692,14 @@ def handler(args: ops.testing.ExecArgs):
         assert args_history[-1].environment == {"foo": "hello", "foobar": "barfoo"}
 
 
-class TestActions(unittest.TestCase):
-    def setUp(self):
+class TestActions:
+    @pytest.fixture
+    def action_results(self):
         action_results: typing.Dict[str, typing.Any] = {}
-        self._action_results = action_results
+        return action_results
```

Comment:
I don't love how this fixture is just a dictionary (and how the `harness` fixture depends on it). Is there a better/simpler way of handling this?

Reply:
Another solution is to extend the `harness` fixture or the client fixture, so that actions can be read from it, something like `assert harness.actions == ...`.

Reply:
OK, I have dropped the initial `action_results.clear()`, I will keep it (instead of on `self`) for now.